### DISCUSSION TITLE
How to build local nuget package?

### DISCUSSION THREAD

**Discussion by timcassell** [dotnet/BenchmarkDotNet#2341] [Q&A] [answered]
Context: I'm trying to test running benchmarks in NativeAOT with the `InProcessNoEmitToolchain`. I attempted to build using csproj references, but the compiler spat errors about `Assets file 'BenchmarkDotNet\src\BenchmarkDotNet.Disassembler.x64\obj\project.assets.json' doesn't have a target for 'net462/win7-x64'.`. I tried building with the nuget reference and it worked (though I got runtime errors). So I'm thinking if I build a local nuget package with my changes and reference that, it should work, but I'm not sure how. Any pointers?

**Comment by timcassell** [comment]
Ah I figured it out. It was just `dotnet pack .\BenchmarkDotNet.sln -c Release`. Then add the `src\BenchmarkDotNet\bin\Release\BenchmarkDotNet.0.13.5-develop.nupkg` and `src\BenchmarkDotNet.Annotations\bin\Release\BenchmarkDotNet.Annotations.0.13.5-develop.nupkg` to my local nuget feed.

**Comment by adamsitnik** [comment]
There is one tricky thing related to local nuget packages: nuget is caching the packages per version. So if you create a `0.13.5-develop` and restore it, then it's going to install the package. But if you apply some changes, run pack again without changing the version number, the next restore won't update the package and you are going to end up using the old version.

**Reply by timcassell** [reply]
Yup, it might be overkill, but I run `dotnet nuget locals all --clear` when I need to rebuild it.

**Comment by AndreyAkinshin** [comment, marked as answer]
@timcassell, the recommended way to pack all the nupkgs is to run `build.[sh/ps1/bat] --target Pack`. This command does the preparatory work and puts the final nupkgs into the `artifacts` folder.
